hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/s3guard/PathMetadata.java

```diff
@@ -91,6 +94,10 @@ public Tristate isEmptyDirectory() {
 
   void setIsEmptyDirectory(Tristate isEmptyDirectory) {
     this.isEmptyDirectory = isEmptyDirectory;
+    if (fileStatus.isDirectory()) {
```

Comment:
Why create a new FileStatus here instead of updating? Suspect we *might* be losing some fields here.

Reply:
Fair question.  Although FileStatus objects are mutable in general, it didn't look to me like the hadoop-aws code base was mutating them after construction anywhere else so I was a little hesitant to do it.  I'll try the change and go ahead with it though so long as tests don't surface any problems.

Reply:
Changed in https://github.com/apache/hadoop/pull/675/commits/3ff59e48785f72c969b5aaacf720743da42a8f3f